e2etests/internal/runner/runner.go

```diff
@@ -358,14 +358,14 @@ func (tm CLI) RunInteractive(ioFunc InteractiveIO, args ...string) RunResult {
 	outFunc := func(r io.Reader) ExpectedRead {
 		return func(want string) {
 			t.Helper()
-			// We try to read more than len(want), assuming its <1024.
-			// So if there is more than the expected output, we will fail.
-			buf := make([]byte, 1024)
-			n, err := r.Read(buf)
+
+			// Try to read in len(want) bytes. Will block while waiting for more output.
+			buf := make([]byte, len(want))
```

Comment:
Consider verifying that there isn't a deadlock if the len < want (legitimiately)